src/tools/rustfmt/src/parse/session.rs

```diff
@@ -231,7 +199,21 @@ impl ParseSess {
     }
 
     pub(crate) fn set_silent_emitter(&mut self) {
-        self.parse_sess.dcx = DiagCtxt::with_emitter(silent_emitter());
+        // Ideally this invocation wouldn't be necessary and the fallback bundle in
+        // `self.parse_sess.dcx` could be used.
+        let fallback_bundle = rustc_errors::fallback_fluent_bundle(
+            rustc_driver::DEFAULT_LOCALE_RESOURCES.to_vec(),
+            false,
+        );
+        // SAFETY: `fatal_dcx` will not be dropped while it is referenced by `self.parse_sess.dcx`,
+        // and `self.parse_sess.dcx` is written to by `ptr::write` so won't drop `fatal_dcx`.
```

Comment:
I'm not too keen on adding unsafe code to rustfmt. To the best of my knowledge we don't have any unsafe code in rustfmt.

Reply:
yeah this would be the very first `unsafe` block that I'm aware of and if there's any way we could avoid it then we'd have a strong preference to do so, even if it results in increased code verbosity

Reply:
Removed the unsafe code now.